docs/design/dp_network_policy_arktos_mizar_int.md

```diff
@@ -0,0 +1,77 @@
+
+## Design Consideration and Challenges  
+
+Previously, with kind cluster, mizar can call kubernete API server directly and access operator store at the same time. However, with arktos integration, arktos API server and network policy operator (in red) does not communication with each other directly. 
+
+* The communication between arktos API server and mizar network policy operator is one directional. Mizar network policy operator cannot send requests back to arktos API server. 
+* As mentioned in section “network policy create/update”, network policy creation process needs access both mizar operator store and arktos API server. If we were to move the entire network policy calculation process (currently managed by network policy operator in red) into network policy controller (in yellow), we can now send request to arktos API server. However, by doing so, we then lost access to mizar operator store and stored endpoint object information. 
+
+## Proposal
+
+We can add couple dictionaries into mizar operator store to keep track of network policy related objects (network policy, pods, and namespaces): 
+1. Key = node’s host ip & value = names of the pods on each node. 
+2. Key = pod name & value = its labels 
+3. Key = namespace name & value = its labels 
+
+Basically, we are trying to avoid making arktos API server calls from network policy operator (in red). Since network policy operator can only access operator store (not arktos API server), we can keep a copy of pods and namespace information in operator store which is necessary for network policy calculation. 
+
+## Key Implementation
+
+### 1. Mizar Operator Store
+
+Add three additional object store into Mizar operator store to track pods and namespaces: 
+* node_pods_store: key = node's host ip address & value = names of the pods on each node
+* pod_label_store: key = pod name & value = each pod's labels
+* namespace_label_store: key = namespace name & value = each namespace's labels
+
+Are we going to have race condition here??? How to avoid race condition
+
+### 2. Add Network Policy Controller in Arktos Repo
+
+* Watch network policy events such as create, update and delete
+* Compose network policy's metadata information into GRPC message and send through GRPC client 
+
+### 3. Add Namespace Controller in Arktos Repo (C2C)
+
+* Watch namespace events such as create, update and delete
+* Compose network policy's metadata information into GRPC message and send through GRPC client
+
+### 3. Add Label Information into GRPC messages for pods and namespaces (C2C)
+* List all labels' key/value sets into GRPC message for pod events
+
+### 4. Remove all kubenetes API Calls During Network Policy Calculation 
+* Remove kubernetes API calls

```

Comment:
Will these calls need to be made on the Arktos side?

Reply:
I don't think you don't need to remove anything on the Mizar side, we still need these API calls when running on vanilla Kubernetes right?

We simply need to restructure the network policy operator in way such that it can function with the new gRPC call.

If there's a function that makes an API call in the Mizar code during network policy creation, we can simply overload that function with an additional flag, or create a separate function to distinguish for Arktos.

Reply:
Discussed offline. The API call being made is simple and can be kept track of in the operator store